install_config/upgrading/migrating_etcd.adoc

```diff
@@ -0,0 +1,274 @@
+to your existing inventory file with both masters and etcd hosts defined in their separate groups. 
+
+. Ensure you have the latest version of the *openshift-ansible* packages
+installed:
++
+----
+# yum upgrade openshift-ansible\*
+----
+
+. Run the *_migrate.yml_* playbook using your inventory file:
++
+----
+# ansible-playbook [-i /path/to/inventory] \
+ifdef::openshift-enterprise[]
+    /usr/share/ansible/openshift-ansible/playbooks/byo/openshift-etcd/migrate.yml
+endif::[]
+ifdef::openshift-origin[]
+    ~/openshift-ansible/playbooks/byo/openshift-etcd/migrate.yml
+endif::[]
+----
+
+[[etcd-data-migration-manual]]
+== Running the Migration Manually
+
+The following procedure describes the steps required to successfully migrate the
+cluster (implemented as part of the Ansible etcd migration playbook).
+
+. +++<b>Stop masters and wait for etcd convergence:</b>+++
+
+.. Stop all master services:
++
+----
+# systemctl stop atomic-openshift-master \
+    atomic-openshift-master-api \
+    atomic-openshift-master-controllers
+----
+
+.. Before the migration can proceed, the etcd cluster must be healthy
+and raft indices of all etcd members must differ by one unit at most.
+At the same time, all etcd members and master daemons must be stopped.
++
+To check the etcd cluster is healthy you can run:
++
+----
+# etcdctl <certificate_details> <endpoint> cluster-health
+member 2a3d833935d9d076 is healthy: got healthy result from https://etcd-test-1:2379
+member a83a3258059fee18 is healthy: got healthy result from https://etcd-test-2:2379
+member 22a9f2ddf18fee5f is healthy: got healthy result from https://etcd-test-3:2379
+cluster is healthy
+----
++
+To check a difference of raft indices you can run:
++
+----
+# ETCDCTL_API=3 etcdctl <certificate details> <endpoint> -w table endpoint status
++------------------+------------------+---------+---------+-----------+-----------+------------+
+|     ENDPOINT     |        ID        | VERSION | DB SIZE | IS LEADER | RAFT TERM | RAFT INDEX |
++------------------+------------------+---------+---------+-----------+-----------+------------+
+| etcd-test-1:2379 | 2a3d833935d9d076 | 3.1.9   | 25 kB   | false     |       415 |        995 |
+| etcd-test-2:2379 | a83a3258059fee18 | 3.1.9   | 25 kB   | true      |       415 |        995 |
+| etcd-test-3:2379 | 22a9f2ddf18fee5f | 3.1.9   | 25 kB   | false     |       415 |        995 |
++------------------+------------------+---------+---------+-----------+-----------+------------+
+----
++
+If minimum and maximum of raft indexes over all etcd members differ by more than
+one unit, wait a minute and try again. Additionally, the migration should not be
+run repeatedly, as new v2 data can overwrite v3 data that has already migrated.
+
+. +++<b>Migrate and scale up etcd:</b>+++
+
+.. Stop etcd on all etcd hosts:
++
+----
+# systemctl stop etcd
+----
+
+.. The migration itself is performed by running the following command (with the
```

Comment:
Run the following command (with the *etcd* daemon stopped) on your first etcd host  to perform the migration: